File: src/main/java/com/kaishiyoku/youtubedownloadhelper/helper/ConsoleHelper.java
package com.kaishiyoku.youtubedownloadhelper.helper;

import org.pmw.tinylog.Logger;

import java.io.IOException;

public class ConsoleHelper {
    public static void cls() {
        System.out.print("\033[H\033[2J");
        System.out.flush();
    }

    public static void render(String str) {
        System.out.println(str);
    }

    public static void println() {
        System.out.println("");
    }

    public static void println(String str) {
        System.out.println(" " + str);
    }

    public static void print(String str) {
        System.out.print(" " + str);
    }

    public static void pressToContinue() {
        println("");
        println("Press any key to continue");

        try {
            System.in.read();
        } catch (IOException e) {
            Logger.error(e);
        }
    }

    public static boolean isWindows() {
        return System.getProperty("os.name").startsWith("Windows");
    }
}

File: src/main/java/com/kaishiyoku/youtubedownloadhelper/helper/ConsoleHelper.kt
package com.kaishiyoku.youtubedownloadhelper.helper

import com.google.gson.Gson
import com.google.gson.GsonBuilder
import org.pmw.tinylog.Logger
import java.io.IOException

object ConsoleHelper {
    fun isWindows(): Boolean = System.getProperty("os.name").startsWith("Windows")

    fun println(str: String) = System.out.println(" $str")

    fun print(str: String) = System.out.print(" $str")

    fun cls() {
        print("\u001b[H\u001b[2J")
        System.out.flush()
    }

    fun printPressToContinue() {
        println("")
        println("Press any key to continue")

        try {
            System.`in`.read()
        } catch (e: IOException) {
            Logger.error(e)
        }
    }

    fun defaultGson(): Gson {
        return GsonBuilder().setPrettyPrinting().create()
    }
}
